### PR TITLE
A red spy has entered the base: Adds Spies, a roundstart antagonist inspired by Goonstation's Spy-Thief

### PR DESCRIPTION
**Pull request by MrMelbert**
# Disclaimer: No Goon code was referenced or used in the making of this PR 

## About The Pull Request

[Design Document (Read this for more information)](https://hackmd.io/@L9JPMsZhRO2wI25rNI6GYg/rkYKM9Yc6)

This PR adds Spies as a new roundstart antagonist type, inspired by Spy-Thiefs from Goonstation.

Spies are tasked with stealing various objects around the station, from insulated gloves to the black box, from the clown's left leg to the bridge's communications console. 

For every item stolen, the Spy is rewarded with a random item from the Syndicate Uplink, plus some items uniquely available to the Spy. Stolen items are then shipped off and sold on the Black Market Uplink, allowing the crew - or maybe some other evil-doers - to get their hands on them. 

image

More ideas for theft items and bounties are welcome. 

## Why It's Good For The Game

See the design document for more information. 

In short: Adds a solo antagonist which has less impact than your Traitors and Heretics, but more impact than Paradox Clones and Thieves. In other words: On the same tier as old traitors. 

Seeks to embrace the sandbox aspect of antagonists more by having no precise greentext objective, and instead some suggestions for chaos you can embark in. Have fun with it!

## Changelog

:cl: Melbert
add: Spies may now roam the halls of Space Station 13. Watch your belongings closely. 
/:cl:

### CODE REVIEW

**Comment by Xkallubar** [comment]
FUN antag?! Epic

**Comment by KingkumaArt** [comment]
Thief walked so this pr could run. I like what I'm hearing - and the black market thing is genius. Also a big fan of how some of this is shit nobody would care if you took at all, like the fax machine, but because it has a reward attached to it, you can get stuff like "John Doe, age 37, has now reached 3rd place on icebox's 'most wanted' list, after stealing the HOS's minifridge".

**Comment by luxidoptera** [comment]
This looks really good. We love a good mischief antag.

**Comment by SomeoneYouProbablyKnow** [comment]
First of all this is wonderful. I love this. A few questions.

How does uplink preference work? What if i want an inplant as traitor and pen as spy? are those just different tabs?

What's the murder policy with this one? That was a big stickiking point for theif?

What's the polocy of taking the possible uplinks of possible spies. Sometimes people do that with traitors, just take away their radio pda and pen. This would be more devastating to a spy and they have no way to make a backup. Maybe they could use the black market uplink itself as a last resort?

>  "John Doe, age 37, has now reached 3rd place on icebox's 'most wanted' list, after stealing the HOS's minifridge"

Speaking of lists. Do roundends list stuff like amount of bounties claimed for spies? That'd be a nice sort of leaderboard.

**Comment by newfren** [comment]
if I have an implant uplink how do I tap an item with it?

**Comment by Jacquerel** [comment]
From reading the code it appears that if your uplink pref is "implant" then it will randomly select between pen or PDA, as this requires a physical item. Perhaps it just shouldn't use the pref at all, although it's mostly harmless.

**Review by Wallemations — APPROVED** [review]
I can't tell what that scan beam is for but I think it's fine idk

**Comment by CCC23** [comment]
finally theif 2

**Comment by scriptis** [comment]
"it shoots bulldogs" i giggled

for historical relevance you should add shoes as an objective, and i'd also suggest the cargo consoles as an objective

one thing that worries me: what if i sign up as an engineer and build fifty fax machines and crewmons in a room in maintenance somewhere? will i have gamed the system? might want to lock building thefts behind roundstart machines only

**Comment by scriptis** [comment]
actually nah, given the variety of machines that'd be a pain in the neck to do. this looks sick

**Comment by ZeWaka** [comment]
It never ends does it

**Comment by carlarctg** [comment]
Love this! Suggestion: An announcement if a high value item is stolen so people know to check out the black market, and how. Time delayed of course.

**Comment by Xkallubar** [comment]
I think this is awesome but the main issue I have with it is the name. 

It's a thief that takes robbery bounties, they don't spy at all. This antag isn't affiliated with any corporation (other than the fact that they are employed by NT) and is just a mercenary/thief, since the Syndicate has Traitors for stuff they want stolen. An independent thief for hire is not a spy at all. Comms agent exists and that's way more of a spy.

A more fitting name would be something like Pariah, Filcher, Crook, Dacoit, Footpad, Cutpurse, Contract Thief, or literally just name it Thief and call it a thief rework.

**Comment by MrMelbert** [comment]
They are Spies because they do secretive heists, have tons of random gadgets, embark upon dangerous missions, and have no discernible ties to any groups or entities outside of their own agency.

**Comment by Timberpoes** [comment]
Sounds like you've created Han Solo to me. Smugglers.

**Comment by HoodMilk** [comment]
Being able to buy the items back from the black market is extremely funny and also solves the issue of them just deleting things from the round. 10/10

**Comment by DustanHache** [comment]
Maybe add plant bug objectives back in on this to make it more thematic. probably as a medium risk objective due to the do after timer.

**Comment by TheVekter** [comment]
I am fucking _rolling_ at the idea of stealing the clown's left leg and finding it on the black market later.

Local clown starts GoFundMe to buy back the leg some cunt stole from him.

**Comment by SamsonTheMVP** [comment]
EXTREMELY overtuned, IMO as a Certified Sec Main. 2 threat antags that are allowed to do anything on LRP and have the antag pass (i.e. lax escalation, can kill sec + command) on MRP. Additionally, they have a nigh-infinite supply of high-cost syndicate agent, nuclear operative, and non-obtainable items (including elite MODsuits, holoparasites, Bulldog Shotguns, energy shield MOD attachments, Ansem pistols, fucking Reticences???). There's the argument to be made that's "They don't have real objectives and are there just to be a small thing to spice up the round!", but this is just not true. Their "main objectives" (i.e. the objectives they're assigned) are things like "Make sure no (department) personnel escape on the shuttle." "Make sure (department) is burned to the ground." - things that are arguably worse than heretics and traitors pre-ascension/final obj. Additionally, I've seen objectives like steal a bodypart from a specific person. So, we've got an almost infinite (if not infinite) ramp antag, that is quite often tasked to sabotage the station in great and horrible ways, getting items that are not meant for players to have normally, with the option to kill more people to get more items? All for TC? This seems a bit overtuned, to say the least.

**Comment by Fmk89** [comment]
Maybe focus their objectives on the number and quality of heists performed, similar to how Changeling DNA strings or Heretic knowledge points are tracked.

Would also be nice to have some sort of progression instead of the random rewards. Maybe gaining knowledge points to spend unlocking crafting recipes and know-how to use custom spy gear focused on breaking and entering and deception. Things like expendable jerry-rigged jaws of life, more convincing cardboard IDs, paper-mache crewman masks, sewing kits to change the appearance of jumpsuits, limited use voice changer devices all sort of expendable MacGyver style items to differentiate the spy from the regular traitors and make their heists more fun/doable.

**Comment by MrMelbert** [comment]
> EXTREMELY overtuned, IMO as a Certified Sec Main. 2 threat antags that are allowed to do anything on LRP and have the antag pass (i.e. lax escalation, can kill sec + command) on MRP. 

To start, have you actually seen a round with them being overtuned are you speaking proactively, and if so, do you have a round ID so I can investigate the logs to see what to tune?

> Additionally, they have a nigh-infinite supply of high-cost syndicate agent, nuclear operative, and non-obtainable items (including elite MODsuits, holoparasites, Bulldog Shotguns, energy shield MOD attachments, Ansem pistols, fucking Reticences???). 

It requires a fair bit of luck and stealing some rather hard to get items to get some of these. For the guns, they're not given extra ammo with the bounty, so once they're done they're done until another bounty rolls that given them ammo. 
Also Reticence was not intended to be in the pool.

> There's the argument to be made that's "They don't have real objectives and are there just to be a small thing to spice up the round!", but this is just not true. Their "main objectives" (i.e. the objectives they're assigned) are things like "Make sure no (department) personnel escape on the shuttle." "Make sure (department) is burned to the ground." - things that are arguably worse than heretics and traitors pre-ascension/final obj. 

The point is that they have to accomplish these tasks with whatever random things they are given, which more often than not is impossible for a majority of players. 
But feel free to suggest more objectives to put in their pool to pad out the more "impactful" ones. 

> Additionally, I've seen objectives like steal a bodypart from a specific person. So, we've got an almost infinite (if not infinite) ramp antag, that is quite often tasked to sabotage the station in great and horrible ways, getting items that are not meant for players to have normally, with the option to kill more people to get more items? All for TC? This seems a bit overtuned, to say the least.

It's not infinite at all. 
- The loot pool drains over time. 
- The bounty pool only refreshes every 10 minutes. 
- One person cannot feasibly use every single item they get as a reward. 

However, once again, if you have seen this occur *in real time*, and are not just speaking off the cuff, please include a round ID or shoot me a ping when it happens so I can analyze it.

**Comment by MrMelbert** [comment]
> Maybe tone down the mayhem and destruction of the Sandbox objectives and limit power and scope of reward items.

Feel free to suggest more sandbox objectives. 
Also feel free to suggest more bounty objectives to match the value of the upper level of items. 

> Would also be nice to have some sort of progression instead of the random rewards.

Uh... no. The random rewards is the entire point of the antag.
See the design document. It's not meant to be a complex progression based antag.
It's meant to be easy to pick up, fun, randomness.

**Comment by SomeoneYouProbablyKnow** [comment]
> For the guns, they're not given extra ammo with the bounty, so once they're done they're done until another bounty rolls that given them ammo.

I do have to inform you hacked lathes print ALL kinds of ammo for some ungodly reason, not just revolver caliber. So getting a reload is not unfeasible. Thankfully very few people know that.

> The random rewards is the entire point of the antag.

Gotta say love random objective - random reward stuff. Too many antags don't have an incentive to improvise and that's where the fun is at.

**Comment by K4rlox** [comment]
i have one question because it seems nobody asked this before

are spies considered "valid"

because as fun as they sound, this will make crew paranoid to people stealing

example: steals captain's fax machine

gets decapitated for "valid"

**Comment by Fmk89** [comment]
> A big issue with Dynamic (at least on MRP) is that we are strapped for low impact antagonists (...)

There is nothing low impact on the current implementation.

**Comment by xzero314** [comment]
Last night I managed to roll spy despite my only selected job being ai on high or return to lobby. It seems spy is not included in that mechanic where you are disqualified for an antag in cases such as that. ALSO In the same round somebody got spy and another antag at the same time.

**Comment by MrMelbert** [comment]
> Last night I managed to roll spy despite my only selected job being ai on high or return to lobby. It seems spy is not included in that mechanic where you are disqualified for an antag in cases such as that. ALSO In the same round somebody got spy and another antag at the same time.

Please include round IDs if you are reporting assignment bugs!!

**Comment by saprasam** [comment]
the loot spies get is fucking hilarious & i hope it only gets more retarded in the future

**Comment by xzero314** [comment]
> > Last night I managed to roll spy despite my only selected job being ai on high or return to lobby. It seems spy is not included in that mechanic where you are disqualified for an antag in cases such as that.
> 
> You got antagonist because you rolled AI successfully but you also rolled antagonist `DEBUG-JOB: Player: Xzero314 is now Rank: AI, JCP:0, JPL:1` `DEBUG-JOB: FOC Error: AI is not compatible with some antagonist role assigned to you., Player: Xzero314`
> 
> > ALSO In the same round somebody got spy and another antag at the same time.
> 
> Please include round IDs if you are reporting assignment bugs!!
> 
> At a glance this seems like you're wrong, no one gained two antag datums, however there was a spy which was converted to blood brother.

The reason why its a bug is the only roll I had selected was ai or return to lobby so this should have disqualified me from rolling spy.   Forget about the 2nd part though I probably misunderstood somebody else saying they had double antag.

**Comment by MrMelbert** [comment]
> The reason why its a bug is the only roll I had selected was ai or return to lobby so this should have disqualified me from rolling spy.

That doesn't solely disqualify you from rolling antag, it just greatly reduces your chance of getting an antag role

**Comment by Dorsisdwarf** [comment]
Had a fun experience with spies the other day, assistants sneaking onto the aisat and scaring all the silicons... then stealing some random stuff and escaping while we went "ohhhhhhhh". There's a certain tense joy in having antags of this level on board, causing trouble and mayhem but like, they generally arent hurting people.

**Comment by MrMelbert** [comment]
This is a problem with the bounty / theft pool being (admittedly) small at the moment, I'm open to suggestions to expand it.

**Comment by Onionmaybe** [comment]
right click on the mech tool on the ground seems to give a wrong reward, and does not take the bounty away.
didnt try to do this multiple times to see if it keeps working

**Comment by MrMelbert** [comment]
> right click on the mech tool on the ground seems to give a wrong reward, and does not take the bounty away. didnt try to do this multiple times to see if it keeps working

Please include more information. Round ID and what reward, namely.

**Comment by Onionmaybe** [comment]
> > right click on the mech tool on the ground seems to give a wrong reward, and does not take the bounty away. didnt try to do this multiple times to see if it keeps working
> 
> Please include more information. Round ID and what reward, namely.

224075 and it gave me a injector module for a modsuit

**Comment by MrMelbert** [comment]
https://tgstation13.org/parsed-logs/manuel/data/logs/2024/02/16/round-224075/uplink.txt The log for future reference
I don't have a ton to go off of and there's no runtimes so if you see it happen again try to get a picture or ahelp it

File: code/game/objects/items/storage/medkit.dm [hunk 1]
```diff
@@ -271,6 +271,24 @@
 		/obj/item/storage/pill_bottle/penacid = 1)
 	generate_items_inside(items_inside,src)
 
+/obj/item/storage/medkit/tactical_lite
+	name = "combat first aid kid"
```
Comment (necromanceranne) [inline code/game/objects/items/storage/medkit.dm:275]: is this a deliberate misspelling or is it meant to say 'kit'?

**Comment by Admiral-Gigaton** [comment]
Hey so like... the items spies send to the black market are kept in the same state the entire time... right?

What's stopping me from buying back the item, det scanning it, and then blasting the spy (besides the spy possibly having 128 TC worth of good gear)

Also, can the spy get kits or bundles?

One more thing: the SC fisher gun has been used, talked about, and probably will get a buff soon, could you make this a more accessible item to spies? It would actually be a viable item for them, making it so they can pop the lights, EMP cams for short periods,and then bust the gear out fast before anyone can get a good read on Rehtt Spy before he punches the wall open and gets away like a batman villain

**Comment by DrAmazing343** [comment]
Det scanning stolen gear sounds like sandbox, which to me == good. Incentivizes less brazen and more thought-out spy play (craft gloves for 4 fucking cloth, it's basically free) which will hopefully create even more strategies to both counter and confirm such play.

**Comment by deathrobotpunch** [comment]
one request, you are able to steal limbs off people by scanning them if there is a bounty for said limb

**Comment by MrMelbert** [comment]
> one request, you are able to steal limbs off people by scanning them if there is a bounty for said limb

image

**Comment by bigfatbananacyclops** [comment]
I tested Spy-thief, i think it has potential but it definitely needs some change.

During my 3 spy rounds ( which wax enough), i easily broke in places and stole stuff necessary to get fully equipped. To the point where I got his grace and carp at the same time. 

I have four suggestions.

I think strong items like his grace,cqc or sleeping carp are fine being part of the rewards, BUT I suggest that they require more effort to be acquired, as in, having to steal two or three items to get one of them. And also being rarer to appear in a list. To complement the rarity this is my second suggestion below:

Adding more reward items to the pool, Im unsure if every single syndicate item is added but.. you could add more. Maybe even a syndicate borg but as a high risk high reward objective.

Thirdly, I strongly suggest focusing the stealing objectives on more valuable, resourceful things the crew relies on. Things that actually impact and affect the crew rather than random consoles that anyone can just rebuild. For example: Physical stacks of minerals, or the ORM and Ore silo. The station strongly relies on them for upgrades, especially science. Or the surgery tables in medbay which nowadays every doctor uses frequently to heal people, like, the cryopod objective is ideal. Because medbay uses it, now imagine having an objective where you steal atleast 5 or 6 medkits from medbay storage. The cargo request console, or surgery tools. This things are ideal objectives, instead of easy objectives of stealing consoles that are rarely used, or an md bot. Spy thief should give a challenge even for a beginner, so they learn to 'steal'.

Lastly, this is i strongly suggest too, but during high pops, I believe the PDA-scan when stealing something should be silent and non-visual. You are going to be constantly watched and you will easily be incriminated, people will still be watching you and realize you are stealing stuff. But the PDA is quite literally screaming "HEY EVERYONE, I AM STEALING", attracting even more attention.

**Comment by MrMelbert** [comment]
> The purpose of thief is to be a lesser antag than traitor 

This is not the purpose of thief 

> Adding more reward items( especially disguising, stealing tools specifically designed for thieving) to the pool

Feel free to suggest more. Almost all uplink items (and then some) items are in the pool. Only some high tier nuke ops loot is not whitelisted. 

> For example: Physical stacks of minerals

This would only work if minerals were actually difficult to access and not in every single department

> the ORM and Ore silo

I considered these but I decided against it. 

> the surgery tables in medbay which nowadays every doctor uses frequently to heal people

Operating computer are in the pool but I felt no need to add surgical tables since they're literally just silver tables that anyone could make. 

There's an argument to be made for stasis beds, and they may come later, but I decided against it for now. 

>Lastly, the PDA-scans when stealing something should be silent and non-visual

Nah. Be more stealthy.

**Comment by bigfatbananacyclops** [comment]
> > The purpose of thief is to be a lesser antag than traitor 
> 
> 
> 
> This is not the purpose of thief. It's supposed to be on par with traitor. 
> 
> 
> 
> > Adding more reward items( especially disguising, stealing tools specifically designed for thieving) to the pool
> 
> 
> 
> Feel free to suggest more. Almost all uplink items (and then some) items are in the pool. Only some high tier nuke ops loot is not whitelisted. 
> 
> 
> 
> > For example: Physical stacks of minerals
> 
> 
> 
> This would only work if minerals were actually difficult to access and not in every single department
> 
> 
> 
> > the ORM and Ore silo
> 
> 
> 
> I considered these but I decided against it. 
> 
> 
> 
> > the surgery tables in medbay which nowadays every doctor uses frequently to heal people
> 
> 
> 
> Operating computer are in the pool but I felt no need to add surgical tables since they're literally just silver tables that anyone could make. 
> 
> 
> 
> There's an argument to be made for stasis beds, and they may come later, but I decided against it for now. 
> 
> 
> 
> >Lastly, the PDA-scans when stealing something should be silent and non-visual
> 
> 
> 
> Nah. Be more stealthy. 
> 
> 
> 
> 
> 
> 

Dear Mr Melbert,

Ore stacks are in fact easy to steal, the consequence is pissing off the crew but i agree it shouldnt be a higher degree of objective.

Regarding the statis beds, if it weren't a single statis bed it would be more impactful.

**Comment by SomeoneYouProbablyKnow** [comment]
> Ore stacks are in fact easy to steal, the consequence is pissing off the crew

Yea it pisses off the crew but it's not high profile enough for people to figure out who did it. Unless you put no effort in hiding it but that's kinda... the whole point of the antag so.
Idk get a disguise, a fake ID, put a signaller on a fuel tank in adjacent maints to distract people, use a hidden away lathe like the engi one, no one ever checks the engi lathe, pay the clown to do it.

File: code/modules/antagonists/spy/spy_bounty.dm [hunk 2]
```diff
@@ -0,0 +1,682 @@
...
+			return pda
+
+	return null
+
+/datum/spy_bounty/targets_person/some_item/pda/target_found(mob/crewmember)
+	. = ..()
+	name = "[crewmember.real_name]'s PDA"
+
+// Steal someone's heirloom
+/datum/spy_bounty/targets_person/some_item/heirloom
```
Comment (Ghommie) [inline code/modules/antagonists/spy/spy_bounty.dm:581]: This should have a `black_market_prob` of 100, very good potential for ransoms.

**Comment by sylvia-from-fulp-station** [comment]
https://github.com/tgstation/tgstation/blob/1de3bb4ea4f4a833a67fff4a637c6f16d24c5289/strings/antagonist_flavor/spy_objective.json#L10
This is incredibly impactful more so than any traitor objective, primary or secondary.  A traitor with die a glorious death could feasibly cause more destruction, and anyone hijacking the shuttle could kill the whole crew, (in the last 3 minutes of the round) but both of those objectives are rare because of their impact.

https://github.com/tgstation/tgstation/blob/1de3bb4ea4f4a833a67fff4a637c6f16d24c5289/strings/antagonist_flavor/spy_objective.json#L9
During a rev round this situation is deemed so unbalanced it ends the game; here it is untracked.

I suggest removing these or changing them to be less impactful but still open to interpretation; eg, 
Ensure all members of x department are; sent to Medical, gravely wounded, victim to a workplace accident

https://github.com/tgstation/tgstation/blob/1de3bb4ea4f4a833a67fff4a637c6f16d24c5289/strings/antagonist_flavor/spy_objective.json#L50-L62
Departments are affected in a few different ways, for their personnel right now it is just 5 different ways to say make sure they don't leave. You CAN interpret this as build a silly little maze to stop them from leaving, but nobody will. I have only seen the objective go un-attempted, or an entire department round removed.
https://github.com/tgstation/tgstation/blob/1de3bb4ea4f4a833a67fff4a637c6f16d24c5289/strings/antagonist_flavor/spy_objective.json#L77-L83
Finally, i think you should experiment with progressive tiers of loot for the steal rewards, getting nuke ops gear isn't possible for anyone other than nuke ops who are some of the rarest antags in the game and it's potentially available to spys roundstart.

**Pull request merged**